test/test.js

```diff
@@ -7,6 +7,14 @@ describe('resolve-dependency-path', function() {
     assert(resolved.indexOf(__dirname) === 0);
   });
 
+  it('resolves with multiple periods in the dependency path', function() {
+    var depPath = './bar.baz.qux';
```

Comment:
This is a bit more complicated than it seems. This module is also used for amd path resolutions (where loader plugins allow mustache, css, and text files as declared dependencies). For those types of dependencies, the `.js` extension will be incorrectly applied. 

Perhaps it's best to resolve the path, then check if a file with that path exists, and if not, find the key/extension in `require.extensions` that results in a found file (defaulting to `.js` if none were found to exist).